engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingPipeline.java

```diff
@@ -45,119 +31,149 @@
 public class ChunkProcessingPipeline {
 
     private static final int NUM_TASK_THREADS = 4;
+    private static final int NUM_CHUNKS_AT_ONCE = 8;
     private static final Logger logger = LoggerFactory.getLogger(ChunkProcessingPipeline.class);
 
+    private final InitialChunkProvider initialChunkProvider;
+    private final Object waitForNewChunks = new Object();
```

Comment:
What is this `Object` used for? I think I've never seen a pure `Object` used anywhere before ... :eyes: 

edit: ah, I see: https://www.baeldung.com/java-wait-notify

Reply:
Yeah, it's just a lock. I suppose it would probably clearer to actually use a Lock.